### DISCUSSION TITLE
Log 'Time' field format

### DISCUSSION THREAD

**Discussion by Mechanolatry** [drakkan/sftpgo#426] [Q&A]
Hi,

Trying to create some dashboards from the logging and trying to understand the log field format - It appears to not be any of the standardized ISO8601, RFC822 or RFC850 formats.

Can you confirm that the datetime format in the logs is consistent - eg. The mm/dd order does not change based upon host system, the timestamp is always UTC or system local time, etc.?

Could a possible enhancement be made to allow the log format to be optionally in the widely used ISO8601 which caters for millisecond precision and removed the ambiguity for all of the above?

So glad I found this project - saved me a lot of work in trying to roll my own containerised sftp with SSH where I was limited by the functionality.

Thanks

**Comment by drakkan** [comment]
Hi,

the log date format is defined [here](https://github.com/drakkan/sftpgo/blob/main/logger/logger.go#L23). It is local time. I think that to make it ISO compliant we should just add the timezone, like this

```
{"level":"info","time":"2021-05-15T10:50:34.431+02:00" .....
```

I'm not sure if this will break some usage (for example fail2ban)

**Reply by drakkan** [reply]
I think the current date time format is already ISO 8601 compliant, according to the ISO 8601 a missing timezone means local time. Please take a look [here](https://www.digi.com/resources/documentation/digidocs/90001437-13/reference/r_iso_8601_date_format.htm) and from the ISO 8601 docs:

> The zone designator is empty if use is made of local time in accordance with 4.2.2.2 through 4.2.2.4, it is the UTC designator [Z] if use is made of UTC of day in accordance with 4.2.4 and it is the difference-component if use is made of local time and the difference from UTC in accordance with 4.2.5.2.
